### ISSUE TITLE
Повторная отправка картриджа на ремонт

### ISSUE THREAD

**Issue by petrkytepov** [sfcl/severcart#139]
Добрый день
Поднимаю вопрос по второй части.
Я выдаю картридж подразделению и они его почти сразу возвращают с дефектом. Как тут это провернуть? Я возвращаю этот картридж в пустые и отправляю на заправку с накопленными или один отдельно. Тут счетчик увеличивать тоже не вижу смысла, так как я этому подразделению выдаю другой заправленный.

**Comment by sfcl** [comment]
Например так. Передаёте пользователю РМ, без фиксации передачи в программе, но сохранив его номер где-нибудь. Если РМ косячный и юзер его сразу возвращает, то передаёте на ремонт контрагенту сразу из страницы (Заправленные). Дальше возвращаете его со всеми или по отдельности с регенерации. При этом можно указать ремонт, без перезаправки. Тогда счётчик перезаправок не увеличится на единицу. Скажу сразу, эта фича реализована в текущей master ветке и данные возможности пока не выкатывались в публичный релиз. Если же РМ исправный, то возвращаете сохраненный номер и фиксируете перемещение. 

Итого: не происходит искажения статистики в случае неисправности РМ.

**Comment by petrkytepov** [comment]
Это понятно, но может пройти пару недель, и тут уже не до сохраненного номера. Объем не большой но по 3-5 картриджей в день приходят.
Вот такой момент. Я не утверждаю что так правильно, но все же. Мы как я уже говорил, используем принцип быстрой заправки, нам принесли картридж в отдел на заправку и мы сразу выдаем подразделению заправленный из запаса. Даже если мне принесли недавно выданный картридж с проблемной печатью, я все равно выдаю другой заправленный картридж. Это сокращает ожидание (бесперебойная печать).
Может Имеет смысл увеличивать счетчик на подразделение в тот момент когда я возвращаю картриджи с заправки? А там в картридже со всеми действиями производимыми с картриджем если отмечу не учитывать счетчик(например), то на подразделение эта заправка не будет учитываться. А история ремонта картриджа соответственно сохранится.
Очень интересно узнать как другие работают в этой сфере (учет и обслуживание картриджей).

**Comment by sem174** [comment]
Сам смысл в том что при передаче картриджа от заправщика к заказчику его приходится проводить либо в ручную либо по ШК. Все картриджи уже есть в актах тогда. И даже если выясняется что картридж заправлен плохо (брак), по актам он уже принят. Тут либо картриджи проверять поголовно пари передаче, и не принимать брак вообще, либо нужно в системе делать некий маркер который будет уже при следующей передаче картриджа в акте передачи маркировать картриджи в списке что они с последней заправки были забракованы. Маркер должен будет автоматически слетать с картриджа после следующего путешествия в сервис. Как то так, и там ещё с отчетами надо думать как это правильно отображать.

**Comment by petrkytepov** [comment]
Мне заправщики прикладывают к каждому картриджу лист проверки и дефект может вылезти не сразу, например через неделю. И скажем они его только заправили, а через неделю у него фотик накрылся. Я как и положено провожу его по программе, возврат передача на ремонт, но вот счетчик увеличивать на подразделение не справедливо в таком случае.
А акт передачи на заправку я сделал в виде таблички в которой заправщик галочками отмечает, что он сделал с картриджем. Просто очень редко бывает, когда именно брак ремонта, где заправщик просто забыл пружинку поправить..... А так обычно замена детали, это и есть ремонт..

**Comment by sem174** [comment]
Можно даже пойти более интересным путём, отмечать бракованный картридж не маркёром а конкретным указанием брака - тонер , вал или чип. К примеру. Далее в актах передачи, следующей заправки список картриджей с браком формировать отдельным списком или листом. И там уже указывать не только номера картриджей и тип брака но и номер и дату актов когда эти картриджи были приняты от заправщиков. Таким образом когда один акт передаётся в заправку в месте с РМ, они будут видеть свои конкретные косяки. А уже созвониться с менеджерами и уточнить информацию по поводу брака не составит труда. А опираясь на действия по браку, можно уже и отчеты правильно формировать.

**Comment by petrkytepov** [comment]
Вот так у меня выглядит акт пришедший от заправщик
https://yadi.sk/i/-mcbFCjF39aM9v

**Comment by petrkytepov** [comment]
Да да отличная идея, но у заправщиков ведется история и они клеят номера и по ним видят. Я сейчас могу посмотреть кто последний его заправлял и ему же отправить.

**Comment by sem174** [comment]
Мои заправщики ничего не клеят, соответственно историю они могут не вести. И опять же у Вас заправщики отмечают Вашу табличку, а у нас они присылают свой акт готовый уже, в котором не пишут номер а просто модель картриджа и что с ним сделано, и тут уже не совсем понятно с каким картриджем конкретно и что делалось.
Так что необходимо расширять функционал системы учёта на такие случаи.

**Comment by petrkytepov** [comment]
Так, нужно их пнуть, пускай номера клеят или с заправщиками проблема, разговор о переходе к другому заправщику уже на начале решает многие вопросы. Это и для Вас будет удобно.
Если они не ведут историю, как тог да они вообще работают. Я могу по запросу получить от заправщика когда и что делалось с картриджем, даже все приходы этого картриджа. Под меня они переделывают и акты и счета. Скажем у нас руководство потребовало отдельно счет-договор на заправку и счет-договор на ремонт с указание запчастей затраченных, без пробрем заправщики (оба) это реализовали.
